src/Temporalio/Bridge/OptionsExtensions.cs

```diff
@@ -166,17 +167,30 @@ internal static class OptionsExtensions
             }
             else if (options.OpenTelemetry != null)
             {
+                if (options.CustomMetricMeter != null)
+                {
+                    throw new ArgumentException(
+                        "Cannot have OpenTelemetry and CustomMetricMeter metrics options");
+                }
                 openTelemetry = scope.Pointer(options.OpenTelemetry.ToInteropOptions(scope));
             }
+            else if (options.CustomMetricMeter != null)
+            {
+                // This object pins itself in memory and is only freed on the Rust side
+                customMeter = new CustomMetricMeter(options.CustomMetricMeter).Ptr;
+            }
             else
             {
                 throw new ArgumentException(
                     "Must have either Prometheus or OpenTelemetry metrics options");
             }
+            // WARNING: It is important that nothing after this point throws, because we have
+            // allocated a pointer for the custom meter which can only be freed on the Rust side
```

Comment:
nit: Doesn't always happen
```suggestion
            // WARNING: It is important that nothing after this point throws, because we may have
            // allocated a pointer for the custom meter which can only be freed on the Rust side
```